components/psst/browser/content/psst_tab_helper.h

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_COMPONENTS_PSST_BROWSER_CONTENT_PSST_TAB_HELPER_H_
+#define BRAVE_COMPONENTS_PSST_BROWSER_CONTENT_PSST_TAB_HELPER_H_
+
+#include "base/component_export.h"
+#include "base/memory/raw_ptr.h"
+#include "base/memory/weak_ptr.h"
+
+#include "brave/components/psst/browser/core/psst_rule.h"
+#include "build/build_config.h"
+#include "components/sessions/core/session_id.h"
+#include "content/public/browser/media_player_id.h"
+#include "content/public/browser/web_contents_observer.h"
+#include "content/public/browser/web_contents_user_data.h"
+
+namespace psst {
+
+class PsstRuleService;
+
+class COMPONENT_EXPORT(PSST_BROWSER_CONTENT) PsstTabHelper
+    : public content::WebContentsObserver,
+      public content::WebContentsUserData<PsstTabHelper> {
+ public:
+  static void MaybeCreateForWebContents(content::WebContents* contents,
+                                        const int32_t world_id);
+  ~PsstTabHelper() override;
+  PsstTabHelper(const PsstTabHelper&) = delete;
+  PsstTabHelper& operator=(const PsstTabHelper&) = delete;
+
+ private:
+  COMPONENT_EXPORT(PSST_BROWSER_CONTENT)
+  PsstTabHelper(content::WebContents*, const int32_t world_id);
+
+  friend class content::WebContentsUserData<PsstTabHelper>;
+
+  // content::WebContentsObserver overrides
+  void DidFinishNavigation(
+      content::NavigationHandle* navigation_handle) override;
+  void DocumentOnLoadCompletedInPrimaryMainFrame() override;
+
+  const int32_t world_id_;
+  raw_ptr<PsstRuleService> psst_service_ = nullptr;  // NOT OWNED
```

Comment:
`const raw_ref<PsstRuleService>`. This can never be null because the tab helper will never get created in the first place if the feature is disabled